comp/trace/agent/agent.go

```diff
@@ -102,6 +104,7 @@ func newAgent(deps dependencies) (Component, error) {
 		ag.config.Object(),
 		ag.telemetryCollector,
 		statsdCl,
+		traceconcentratorimpl.NewConcentrator(ag.config, nil, time.Now(), statsdCl),
```

Comment:
Here, we either have to create the concentrator here, as I'm doing, or add it as a dep.

Unfortunately, creating it here means it can't be passed in as a generic component, meaning we don't gain anything at all from using components.

Adding it as a dependency means it'll have to be constructed by FX.